CMSgov/cmcs-eregulations
# EREGCSC-1594 -- Fix Vue shorthand errors

Resolves [EREGCSC-1594](https://jiraent.cms.gov/browse/EREGCSC-1594)

**Description**
The `RelatedRuleList` component was using incorrect named slot shorthand syntax and causing an issue where the Show/Hide buttons were not rendering correctly.

See: https://v2.vuejs.org/v2/guide/components-slots.html#Named-Slots-Shorthand

**This pull request changes**

- Removes colons from v-slot shorthand 
- Adds cypress test to ensure buttons are properly labeled

**Steps to manually verify this change**

1. Visit any Subpart in reader view
2. Expand FR Docs category in right sidebar
3. Make sure the buttons to expand and collapse the FR docs is labeled "+ Show More (X)" and "- Show Less (X)"
